model/project_ref.go

```diff
@@ -1198,11 +1197,10 @@ func GetTasksWithOptions(projectName string, taskName string, opts GetProjectTas
 	}
 	match["$and"] = []bson.M{
 		{task.RevisionOrderNumberKey: bson.M{"$lte": startingRevision}},
-		{task.RevisionOrderNumberKey: bson.M{"$gte": startingRevision - tasksByProjectMaxNumVersions}},
+		{task.RevisionOrderNumberKey: bson.M{"$gte": startingRevision - opts.Limit}},
```

Comment:
The documentation says "the number of versions to be returned" but here it's actually that these are the number of versions we're checking (it could be that some versions don't have the task or something in which case we'd skip those versions).

I can't think of a way to actually return the latest 20 versions without _also_ querying the version collection though so that does get maybe a bit expensive; perhaps we should change the documentation (and update the user) that it'll be the limit for what we search rather than for what we return 🤔

Reply:
Also need to update the function comment.

Reply:
I couldn't figure out an aggregation that would keep the behavior that we described so I changed it in the docs to searching the number of versions specified. https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#get-tasks-for-a-project

I don't think this would be an issue because the current behavior wasn't as described so it shouldn't impact workflow. Will update the ticket after merging

Reply:
also updated function comment